src/common-types.ts

```diff
@@ -0,0 +1,342 @@
+/**
+ * Represents a 2-dimensional vector with x and y coordinates.
+ */
+export class Vector2 {
+  /**
+   * The x-coordinate of the vector.
+   */
+  private _x: number;
+
+  /**
+   * Gets the x-coordinate of the vector.
+   */
+  public get x(): number {
+    return this._x;
+  }
```

Comment:
I don't think this will matter until we get to pixels, but it's worth noting that this style of property handling is slower than just directly accessing `_x` by making it public and renaming it to `x`. This is because the getter method goes on the function stack. JS runtimes are really optimized for this case, so the difference doesn't really show until you're dealing with millions of lookups, but with a 512x512 canvas, that'll be just around the number of bytes we have in our pixel data.